gateway-server/src/main/java/org/apache/knox/gateway/websockets/ProxyWebSocketAdapter.java

```diff
@@ -64,7 +65,7 @@
   /* Message buffer for holding data frames temporarily in memory till connection is setup.
    Keeping the max size of the buffer as 100 messages for now. */
   private List<String> messageBuffer = new ArrayList<String>();
-  private static final int MAX_BUFFER_MESSAGE_COUNT = 100;
+  private int maxMessageBufferCount = GatewayConfigImpl.DEFAULT_WEBSOCKET_MAX_WAIT_BUFFER_COUNT;
```

Comment:
I'd remove this class member; instead, you should use `config.getWebsocketMaxWaitBufferCount()` in line 242.

Reply:
How can config object (GatewayConfigImpl class) be accessed in ProxyWebSocketAdapter ?

Reply:
You can have a new final class member; initialized when creating the instance (through the constructor).

Reply:
Yeah passing config as constructor argument makes more sense. Fixed now in new commit.

Reply:
Thanks!